extensions/oidc/deployment/src/main/java/io/quarkus/oidc/deployment/OidcBuildStep.java

```diff
@@ -399,6 +417,80 @@ RunTimeConfigBuilderBuildItem useOidcTenantDefaultIdConfigBuilder() {
+                ai -> {
+                    AnnotationValue maxAgeValue = ai.value("maxAge");
+                    final long maxAge = maxAgeValue != null ? maxAgeValue.asInt() : AuthenticationContext.NO_MAX_AGE;
+
+                    String acrValues = "";
+                    AnnotationValue annotationValue = ai.value();
+                    String[] annotationValues = annotationValue == null ? null : annotationValue.asStringArray();
+                    if ((annotationValues == null || annotationValues.length == 0)
+                            && maxAge == AuthenticationContext.NO_MAX_AGE) {
+                        // no acr values and no max age
+                        throw new ConfigurationException("Annotation '" + AUTHENTICATION_CONTEXT_NAME + "' placed on '"
+                                + toTargetName(ai.target()) + "' specifies neither 'maxAge' or 'acr' values");
+                    } else if (annotationValues != null && annotationValues.length > 0) {
+                        acrValues = String.join(",", annotationValues);
+                    }
+
+                    return acrValues + ACR_VALUES_TO_MAX_AGE_SEPARATOR + maxAge;
+                }, true, AUTHENTICATION_CONTEXT_NAME));
+
+        // @AuthenticationContext -> authentication required
+        // register @Authenticated for annotated methods
+        Set<MethodInfo> annotatedMethods = collectMethodsWithoutRbacAnnotation(authCtxAnnotations
+                .stream()
+                .map(AnnotationInstance::target)
+                .filter(at -> at.kind() == METHOD)
+                .map(AnnotationTarget::asMethod)
+                .toList());
+        additionalSecuredMethodsProducer
+                .produce(new AdditionalSecuredMethodsBuildItem(annotatedMethods, Optional.of(List.of("**"))));
+        // method-level security; this registers @Authenticated if no RBAC is explicitly declared
+        Predicate<ClassInfo> useClassLevelSecurity = useClassLevelSecurity(classSecurityAnnotations);
+        Set<MethodInfo> annotatedClassMethods = collectClassMethodsWithoutRbacAnnotation(
+                collectAnnotatedClasses(authCtxAnnotations, Predicate.not(useClassLevelSecurity)));
+        additionalSecuredMethodsProducer
+                .produce(new AdditionalSecuredMethodsBuildItem(annotatedClassMethods, Optional.of(List.of("**"))));
+        // class-level security; this registers @Authenticated if no RBAC is explicitly declared
+        collectAnnotatedClasses(authCtxAnnotations, useClassLevelSecurity).stream()
+                .filter(Predicate.not(HttpSecurityUtils::hasSecurityAnnotation))
+                .forEach(c -> registerClassSecurityCheckProducer.produce(
+                        new RegisterClassSecurityCheckBuildItem(c.name(), AnnotationInstance
+                                .builder(Authenticated.class).buildWithTarget(c))));
+    }
+
+    private static boolean areEagerSecInterceptorsSupported(Capabilities capabilities,
```

Comment:
I propose just to create a few tests in the `deployment` or a dedicated integration test module, I'd be concerned that if we had this code here then it would stay forever.

Reply:
As you wish, but it makes things harder and I'll have to drop validation for `@Tenant` annotation again, because before this PR, I didn't throw exception for `@Tenant` when proactive auth was enabled just to allow to have tests in the OIDC wiremock. I'll try to create new tests for `@AuthenticationContext` and drop this.

Reply:
At one point I was thinking about introducing way to add custom claims (like `acr`) via OIDC Dev Service, I think that would make move these tests out easy. I'll think about that again.

Reply:
@michalvavrik  Sure, but let's deal with the the dev service enhancement later. Having a default tenant test would do to get started

Reply:
I have dropped this.